### PR TITLE
RDoc-2882 ETL to Azure Queue Storage

### PR DESCRIPTION
**Pull request by Danielle9897**
**Related issue:**
https://issues.hibernatingrhinos.com/issue/RDoc-2882/Document-ETL-to-Azure-Queue-Storage

**Scope of work** (create new article / add new info / fix...):

- Fix the duplicate 'general info' article
- Ongoing tasks operations article (required due to above bullet)
- Ongoing tasks overview article (under Studio)
- AQS ETL article (under Studio)
- Queue ETL overview article (under Server)
- ETL basics article
- AQS ETL article (under Server)
- Update Kafka & RabbitMQ articles (under Server) to match + Place common content in overview
- ETL configuration article (under Server)
- How to add a connection string
- Get connection string
- Remove connection string
- Add ETL operation

### CODE REVIEW

File: Documentation/6.1/Raven.Documentation.Pages/server/ongoing-tasks/etl/queue-etl/rabbit-mq.markdown [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    When replacing the exchange name with an empty string,  
+    the message will be routed using the routingKey via the default exchange, which is predefined by the broker.
+
+  * `loadTo<ExchangeName>(obj, {attributes})`  
+    When omitting the routingKey, messages delivery will depend on the exchange type.
+
+  * `loadTo<ExchangeName>(obj, 'routingKey')`  
+    When omitting the attributes, default attribute values will be assigned.
+
+  * `loadTo<ExchangeName>(obj)`
```
Comment (Danielle9897) [inline Documentation/6.1/Raven.Documentation.Pages/server/ongoing-tasks/etl/queue-etl/rabbit-mq.markdown:148]: @djordjedjukic
What should be written/explained for this overload ?

File: Documentation/6.1/Raven.Documentation.Pages/server/ongoing-tasks/etl/queue-etl/azure-queue.markdown [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    * Note: the following connection string parameters are mandatory:  
+       * `AccountName`
+       * `DefaultEndpointsProtocol`
+       * `QueueEndpoint` (when using http protocol)
+* **Entra ID**
+    * Use the Entra ID authorization method to achieve enhanced security by leveraging Microsoft Entra’s robust identity solutions.  
+    * This approach minimizes the risks associated with exposed credentials commonly found in connection strings and enables
+      more granular control through [Role-Based Access Controls](https://learn.microsoft.com/en-us/azure/role-based-access-control/).
+* **Paswordless**
+    * This authorization method authorizes a dedicated machine and can only be used in self-hosted mode.
```
Comment (poissoncorp) [inline Documentation/6.1/Raven.Documentation.Pages/server/ongoing-tasks/etl/queue-etl/azure-queue.markdown:69]: this authorization method required the machine to be already authorized

**Review by poissoncorp — APPROVED** [review]
i've left just a single comment over the authentication methods in AQS

File: Documentation/6.1/Raven.Documentation.Pages/server/ongoing-tasks/etl/queue-etl/azure-queue.markdown (hunk 2 above)
Comment (Danielle9897) [reply Documentation/6.1/Raven.Documentation.Pages/server/ongoing-tasks/etl/queue-etl/azure-queue.markdown:69]: Modified to:
`This authorization method requires the machine to be pre-authorized and can only be used in self-hosted mode.`
Hope this is ok.

**Review by poissoncorp — APPROVED** [review]

**Review by djordjedjukic — APPROVED** [review]

File: Documentation/6.1/Raven.Documentation.Pages/server/ongoing-tasks/etl/queue-etl/azure-queue.markdown [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+   | **attributes** | object | An object with optional & required [CloudEvents attributes](../../../../server/ongoing-tasks/etl/queue-etl/overview#cloudevents) |
+
+For example, the following two calls, which load data to "OrdersQueue", are equivalent:
+
+* `loadToOrdersQueue(obj, {attributes})`
+* `loadTo('OrdersQueue', obj, {attributes})`
+
+---
+
+A sample script that process documents from the Orders collection:
```
[the remark this answers is not in the document]
Comment (Danielle9897) [reply Documentation/6.1/Raven.Documentation.Pages/server/ongoing-tasks/etl/queue-etl/azure-queue.markdown:163]: done

File: Documentation/6.1/Raven.Documentation.Pages/server/ongoing-tasks/etl/queue-etl/azure-queue.markdown [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+---
+
+* This article focuses on how to create an Azure Queue Storage ETL task using the Client API.  
+  To define an Azure Queue Storage ETL task from the Studio, see [Studio: Azure Queue Storage ETL Task](../../../../studio/database/tasks/ongoing-tasks/azure-queue-storage-etl).  
+  For an **overview of Queue ETL tasks**, see [Queue ETL tasks overview](../../../../server/ongoing-tasks/etl/queue-etl/overview).
+
+* In this page:  
+  * [Add an Azure Queue Storage connection string](../../../../server/ongoing-tasks/etl/queue-etl/azure-queue#add-an-azure-queue-storage-connection-string)  
+      * [Authentication methods](../../../../server/ongoing-tasks/etl/queue-etl/azure-queue#authentication-methods)  
+      * [Exmaple](../../../../server/ongoing-tasks/etl/queue-etl/azure-queue#example)
```
[the remark this answers is not in the document]
Comment (Danielle9897) [reply Documentation/6.1/Raven.Documentation.Pages/server/ongoing-tasks/etl/queue-etl/azure-queue.markdown:34]: done

**Review by arekpalinski — APPROVED** [review]

**Comment by Danielle9897** [comment]
@djordjedjukic 
so I added the `AccountKey` to the list of mandatory parms for the AQS Connection String in the latest commit here
(as @kalczur has checked its necessity in [this pr comment](https://github.com/ravendb/ravendb/pull/18919#discussion_r1690871731)

**but... now I wonder...**
shouldn't the code [here](https://github.com/ravendb/ravendb/blob/6ae5d69e9888ff917db0e1167544ca0abf549013/src/Raven.Client/Documents/Operations/ETL/Queue/AzureQueueStorageConnectionSettings.cs#L65) add a check that this param is provided ?
why isn't it checked there ?

**Comment by djordjedjukic** [comment]
> @djordjedjukic so I added the `AccountKey` to the list of mandatory parms for the AQS Connection String in the latest commit here (as @kalczur has checked its necessity in [this pr comment](https://github.com/ravendb/ravendb/pull/18919#discussion_r1690871731)
> 
> **but... now I wonder...** shouldn't the code [here](https://github.com/ravendb/ravendb/blob/6ae5d69e9888ff917db0e1167544ca0abf549013/src/Raven.Client/Documents/Operations/ETL/Queue/AzureQueueStorageConnectionSettings.cs#L65) (or in any other place) add a check that this param is provided ?

AccountKey is included in the Connection string. The user doesn't input that separately. You can see that we are parsing the connection string using the method you provided.

If that is what you are asking?

**Review by djordjedjukic — APPROVED** [review]

**Review by arekpalinski — APPROVED** [review]

**Pull request merged**